crates/subspace-runtime/src/lib.rs

```diff
@@ -1115,6 +1120,10 @@ impl_runtime_apis! {
         fn receipt_hash(domain_id: DomainId, domain_number: DomainNumber) -> Option<DomainHash> {
             Domains::receipt_hash(domain_id, domain_number)
         }
+
+        fn domain_transaction_byte_fee() -> Balance {
+            DOMAIN_STORAGE_FEE_MULTIPLIER * TransactionFees::transaction_byte_fee()
```

Comment:
Looks like this `transaction_byte_fee` returns the next block transaction byte fee instead of current block since the value is cleared at finaliation. Ideally, we would want the byte fee to be sent as inherent to domain that was used for calculate the bundle. So instead of clearing it on finalisation, we should clear on next block initialization instead

Reply:
> Ideally, we would want the byte fee to be sent as inherent to domain that was used for calculate the bundle. So instead of clearing it on finalization, we should clear on next block initialization instead

Not really, what we want is the byte fee that is used to calculate the bundle in the *next* block because the new byte fee takes effect in the domain chain on block finalization. In this way, we are using the same byte fee for both charging bundles, validating domain tx, and executing domain tx.

Reply:
Hmm not sure I understand this. This is my current understanding. PLease let me know what I'm missing here

Consensus chain uses `transaction_byte_fee` during block #x. All the bundles in that block are changed using the same `transaction_byte_fee`. So once the block is imported by consensus chain, domain chain will start processing the consensus chain block #x. So at this point, it would take the `transaction_byte_fee` consensus chain used and multiplier, which is `2`, and then inject that to domain block as inherent. During domain block state transition, we use the same transaction_byte_fee * multipler to collect fees from the extrinsic that is part of the bundle.